sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala

```diff
@@ -36,6 +36,18 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
   /**
    * For each node, extract properties in the form of a list ['key1', 'key2', 'key3', 'value']
    * into a pair (key1.key2.key3, value).
+   *
```

Comment:
That happens when we have a dot-separated key, e.g.: `a.b.c` The parser identifies each part of the identifier separately, e.g: `a`, `b` & `c`. It is a bit of a PITA  to re-construct this in ANTLR3, so we currently do this in the tree parser.